code/modules/mob/living/simple_animal/friendly/farm_animals.dm

```diff
@@ -339,3 +339,59 @@
 	name = "emperor penguin"
 	desc = "Emperor of all he surveys."
 
+/mob/living/simple_animal/pig
+	name = "pig"
+	desc = "Known for being the main ingrediant in bacon."
+	icon_state = "pig"
+	icon_living = "pig"
+	icon_dead = "pig_dead"
+	speak = list("OINKKK?","Oink!","RRRRRREEEEEEEEEE!")
+	speak_emote = list("Oinks","Oinks hauntingly")
+	emote_hear = list("squeels")
+	emote_see = list("shakes its head")
+	speak_chance = 1
+	turns_per_move = 5
+	see_in_dark = 6
+	meat_type = /obj/item/weapon/reagent_containers/food/snacks/meat/pig
+	meat_amount = 15 //Pigs are fat as fuck, as they grow this will increase
+	response_help  = "pets"
+	response_disarm = "gently pushes aside"
+	response_harm   = "kicks"
+	attacktext = "kicked"
+	health = 280
+	autoseek_food = 0
+	beg_for_food = 0
+	var/datum/reagents/udder = null
+	var/amount_grown = 0
+	mob_size = 20//based on mass of holstein fresian dairy cattle, what the sprite is based on
+
+/mob/living/simple_animal/pig/Initialize()
+	. = ..()
+	var/gender = pick(MALE, FEMALE)
+
+	if(gender == FEMALE) // You cant milk boy pigs you pervs
+		udder = new(50)
+		udder.my_atom = src
+
+/mob/living/simple_animal/pig/attackby(var/obj/item/O as obj, var/mob/user as mob)

```

Comment:
This would be significantly better off being a subtype of cow, rather than just copy pasting the same code.

Reply:
To add onto it - Pigs are a shitload harder to milk than cows. And give off less milk in general. There's a reason we don't drink pig milk instead of cow milk.

Even more - Pigs *really* don't like people touching them down there. or being restrained. really milking should either make them run away, or get them pissed at the person trying to milk them.